pkgs/node/src/utils.zig

```diff
@@ -0,0 +1,32 @@
+const std = @import("std");
+const Thread = std.Thread;
+const Mutex = Thread.Mutex;
+
+const xev = @import("xev");
+
+pub const EventLoop = struct {
+    loop: xev.Loop,
+    // events from libp2p or other threads will also be pushed on it
+    mutex: Mutex,
+
+    const Self = @This();
+    pub fn init() !Self {
+        const loop = try xev.Loop.init(.{});
+        const mutex = Mutex{};
```

Comment:
this mutex doesn't seem like it's being used at all :thinking:

Reply:
yes other events from other theads might also go into this loop